### PR TITLE
added laneID and Lane Position for Objects in Frames

### PR DESCRIPTION
**Pull request by mapohivi**

### CODE REVIEW

File: aveas_openlabel/attributes/open_drive.py [hunk 1]
```diff
@@ -37,6 +37,28 @@ class OpenDrive__RoadId(TextData):
...
 
 
+@dataclass
+class OpenDrive__LaneId(TextData):
+    """Lane ID the object is moving on corresponding to the Road ID (stated in OpenDrive__RoadId) in the OpenDRIVE file"""
+
+    val: str = field(default_factory=lambda: no_default(field="OpenDrive__LaneId.val"), metadata=required)
+    """The OpenDRIVE lane ID"""
+
+    name: Literal["used_lane_id"] = field(default="used_lane_id")
```
Comment (monoclecat) [inline aveas_openlabel/attributes/open_drive.py]: The value has to be `open_drive/lane_id`

File: aveas_openlabel/attributes/open_drive.py [hunk 2]
```diff
@@ -37,6 +37,28 @@ class OpenDrive__RoadId(TextData):
...
+
+
+@dataclass
+class OpenDrive__LanePosition(TextData):
+    """Relatove Lane Position of the object moving on the corresponding Lane ID (stated in OpenDrive__LaneId) in the OpenDRIVE file. Range between -0.5 (right border) and 0.5 (left border)"""
+
+    val: str = field(default_factory=lambda: no_default(field="OpenDrive__LanePosition.val"), metadata=required)
+    """Relative lane position of object on OpenDRIVE lane"""
+
+    name: Literal["used_relative_lane_position"] = field(default="used_relative_lane_position")
```
Comment (monoclecat) [inline aveas_openlabel/attributes/open_drive.py]: The value has to be `open_drive/lane_position`

File: aveas_openlabel/attributes/open_drive.py [hunk 3]
```diff
@@ -37,6 +37,28 @@ class OpenDrive__RoadId(TextData):
     """Is always 'open_drive/road_id'"""
 
 
+@dataclass
+class OpenDrive__LaneId(TextData):
+    """Lane ID the object is moving on corresponding to the Road ID (stated in OpenDrive__RoadId) in the OpenDRIVE file"""
```
Comment (monoclecat) [inline aveas_openlabel/attributes/open_drive.py]: Add backticks (`) to OpenDrive__RoadId so that the documentation creates a hyperlink to the class.

Comment (monoclecat) [inline aveas_openlabel/attributes/open_drive.py]:
```suggestion
    """Lane ID the object is moving on, corresponding to the Road ID (stated in OpenDrive__RoadId) in the OpenDRIVE file"""
```

File: aveas_openlabel/attributes/open_drive.py [hunk 4]
```diff
@@ -37,6 +37,28 @@ class OpenDrive__RoadId(TextData):
...
+    val: str = field(default_factory=lambda: no_default(field="OpenDrive__LaneId.val"), metadata=required)
+    """The OpenDRIVE lane ID"""
+
+    name: Literal["used_lane_id"] = field(default="used_lane_id")
+    """Is always 'open_drive/lane_id'"""
+
+
+@dataclass
+class OpenDrive__LanePosition(TextData):
+    """Relatove Lane Position of the object moving on the corresponding Lane ID (stated in OpenDrive__LaneId) in the OpenDRIVE file. Range between -0.5 (right border) and 0.5 (left border)"""
```
Comment (monoclecat) [inline aveas_openlabel/attributes/open_drive.py]:
```suggestion
    """Relative lane position of the object which is moving on the corresponding Lane ID (stated in `OpenDrive__LaneId`) in the OpenDRIVE file. Range between -0.5 (right border) and 0.5 (left border)"""
```

Take note of the added backticks.

Comment (monoclecat) [inline aveas_openlabel/attributes/open_drive.py]: > Range between -0.5 (right border) and 0.5 (left border)

Please add validation for this as is done for `Interior__BrakePedal`, for example.

File: aveas_openlabel/attributes/open_drive.py [hunk 5]
```diff
@@ -37,6 +37,28 @@ class OpenDrive__RoadId(TextData):
...
+
+    val: str = field(default_factory=lambda: no_default(field="OpenDrive__LanePosition.val"), metadata=required)
+    """Relative lane position of object on OpenDRIVE lane"""
```
Comment (monoclecat) [inline aveas_openlabel/attributes/open_drive.py]:
```suggestion
    """Relative lane position of object on the OpenDRIVE lane"""
```

**Review by monoclecat — CHANGES_REQUESTED** [review]

File: aveas_openlabel/attributes/open_drive.py (hunk 4 above)
Comment (mapohivi) [reply aveas_openlabel/attributes/open_drive.py]: geändert

File: aveas_openlabel/attributes/open_drive.py (hunk 5 above)
Comment (mapohivi) [reply aveas_openlabel/attributes/open_drive.py]: geändert

File: aveas_openlabel/attributes/open_drive.py (hunk 4 above)
Comment (mapohivi) [reply aveas_openlabel/attributes/open_drive.py]: geändert

File: aveas_openlabel/attributes/open_drive.py (hunk 3 above)
Comment (mapohivi) [reply aveas_openlabel/attributes/open_drive.py]: geändert

Comment (mapohivi) [reply aveas_openlabel/attributes/open_drive.py]: geändert und zusätzlich angewendet in OpenDrive__LanePosition und

File: aveas_openlabel/attributes/open_drive.py (hunk 2 above)
Comment (mapohivi) [reply aveas_openlabel/attributes/open_drive.py]: geändert

File: aveas_openlabel/attributes/open_drive.py (hunk 1 above)
Comment (mapohivi) [reply aveas_openlabel/attributes/open_drive.py]: geändert

File: aveas_openlabel/attributes/open_drive.py (hunk 4 above)
Comment (mapohivi) [reply aveas_openlabel/attributes/open_drive.py]: additionally: dataclass changed to numeric

File: aveas_openlabel/attributes/summary.py [hunk 6]
```diff
@@ -258,3 +258,14 @@ class Summary__Coordinates__ScenarioEnd__UStdDev(VectorData):
 
     name: Literal["summary/coordinates/scenario_end/ustddev"] = field(default="summary/coordinates/scenario_end/ustddev")
     """Is always 'summary/coordinates/scenario_end/ustddev'."""
+
+
+@dataclass
+class Summary__Frame__Appearance(VectorData):
```
Comment (monoclecat) [inline aveas_openlabel/attributes/summary.py:264]: This is already covered by the OpenLABEL frame intervals. Why define an attribute for it?

File: aveas_openlabel/object_data/unsteerable_unattached_non_operator.py [hunk 7]
```diff
@@ -78,12 +79,21 @@ class ObjectData__Unsteerable_Unattached_NonOperator(BaseObjectData, EachAttribu
...
+
+
+=======
     text: list[
         Union[
             Impact__Frame,
             Operator__Gender,
         ]
     ] = field(
+>>>>>>> .theirs
```
Comment (monoclecat) [inline aveas_openlabel/object_data/unsteerable_unattached_non_operator.py:96]: This is from a failed merge attempt and needs to be cleaned up

File: aveas_openlabel/__init__.py [hunk 8]
```diff
@@ -102,6 +102,7 @@
     `Summary__Coordinates__ScenarioStart`           **O**     **O**      **O**  **O**  **O**              **O**             **O**         **O**          **O**               **O**          **O**      **O**    **O**
     `Summary__Coordinates__ScenarioStart__UStdDev`  **O**     **O**      **O**  **O**  **O**              **O**             **O**         **O**          **O**               **O**          **O**      **O**    **O**
     `Summary__Coordinates__ScenarioEnd`             **O**     **O**      **O**  **O**  **O**              **O**             **O**         **O**          **O**               **O**          **O**      **O**    **O**
+    `Summary__Frame__appearance`                    **O**     **O**      **O**  **O**  **O**              **O**             **O**         **O**          **O**               **O**          **O**      **O**    **O**
```
Comment (monoclecat) [inline aveas_openlabel/__init__.py]: You forgot to remove this line

**Review by monoclecat — APPROVED** [review]

**Pull request merged**